### PR TITLE
Search functionality: Mentee : Katha 1065

### PR DESCRIPTION
**Pull request by adithyadinesh0412**

### CODE REVIEW

File: src/generics/utils.js [hunk 1]
```diff
@@ -520,7 +524,62 @@ function validateFilters(input, validationData, modelName) {
...
+		}
 
+		let filter = ''
+		if (userPolicyDetails.external_mentor_visibility && userPolicyDetails.organization_id) {
+			// fetch organisation details to get the related org
+			let userOrgDetails = await userRequests.fetchDefaultOrgDetails(userPolicyDetails.organization_id)
+
+			// list of related org ids
+			let orgToList = userOrgDetails.data.result.related_orgs
+			orgToList.push(userPolicyDetails.organization_id)
```
Comment (priyanka-TL) [inline src/generics/utils.js:549]: please change the variable name orgToList

File: src/integration-tests/commonTests.js [hunk 2]
```diff
@@ -12,46 +13,45 @@ const logIn = async () => {
...
+			timeout: 60000,
 		}
 		await waitOn(opts)
-		let email = 'nevil' + crypto.randomBytes(5).toString('hex') + '@tunerlabs.com'
+		let firstname = 'adithya'
+		let email = firstname + crypto.randomBytes(5).toString('hex') + '@tunerlabs.com'
 		let password = faker.internet.password()
 		let res = await request.post('/user/v1/account/create').send({
-			name: 'Nevil',
+			name: firstname,
```
Comment (priyanka-TL) [inline src/integration-tests/commonTests.js:24]: @adithyadinesh0412 directly assign the data first name, so don't want to assign variable

File: src/integration-tests/mentees/mentees.spec.js [hunk 3]
```diff
@@ -6,38 +6,32 @@
...
-		let res = await request.get('/mentoring/v1/mentees/profile')
+		loginInfo = await commonHelper.logIn()
+		await request.post('/mentoring/v1/profile/create').send({
+			designation: ['beo', 'deo', 'testt'],
+			area_of_expertise: ['educational_leadership', 'sqaa'],
+			education_qualification: 'MBA',
+			tags: ['Experienced', 'Technical'],
+			visibility: 'visible',
+			organisation_ids: [1],
+			external_session_visibility: 'limited',
```
Comment (priyanka-TL) [inline src/integration-tests/mentees/mentees.spec.js:21]: @adithyadinesh0412 currently defined policied are CURRENT< ASSOCIATED and ALL. Please use that

File: src/services/mentees.js [hunk 4]
```diff
@@ -758,4 +758,162 @@ module.exports = class MenteesHelper {
...
+					},
+				})
+			}
+			const menteeIds = extensionDetails.data.map((item) => item.user_id)
+
+			if (menteeIds) {
+				userServiceQueries['user_ids'] = menteeIds
+			}
+
+			const userDetails = await userRequests.search(userType, pageNo, pageSize, searchText, userServiceQueries)
```
Comment (priyanka-TL) [inline src/services/mentees.js:830]: @adithyadinesh0412 make sure all function which is using search API also have these changes

File: src/services/mentees.js [hunk 5]
```diff
@@ -758,4 +758,170 @@ module.exports = class MenteesHelper {
...
+						return value
+					}
+					return null
+				})
+				.filter((value) => value !== null)
+
+			let foundKeys = {}
+			let result = []
+			// update count after filters
+			userDetails.data.result.count = userDetails.data.result.data.length
```
Comment (rakeshSgr) [inline src/services/mentees.js:897]: @adithyadinesh0412 count should be total data in the DB.

File: src/services/mentees.js [hunk 6]
```diff
@@ -758,4 +758,170 @@ module.exports = class MenteesHelper {
...
+			extensionDetails = await menteeQueries.getUsersByUserIdsFromView(
+				userDetails.data.result.data.map((item) => item.id),
+				null,
+				null,
+				filteredQuery,
+				saasFilter,
+				additionalProjectionString,
+				false
+			)
+			if (organization_ids.length > 0) {
```
Comment (rakeshSgr) [inline src/services/mentees.js:854]: @adithyadinesh0412 add filters in query

File: src/services/mentees.js [hunk 7]
```diff
@@ -758,4 +758,170 @@ module.exports = class MenteesHelper {
...
+					return null
+				})
+				.filter((value) => value !== null)
+
+			let foundKeys = {}
+			let result = []
+			// update count after filters
+			userDetails.data.result.count = userDetails.data.result.data.length
+
+			for (let user of userDetails.data.result.data) {
```
Comment (rakeshSgr) [inline src/services/mentees.js:899]: @adithyadinesh0412 we don't require this i think

File: src/services/mentees.js [hunk 8]
```diff
@@ -758,4 +758,170 @@ module.exports = class MenteesHelper {
...
+						values: [user],
+					})
+					foundKeys[firstChar] = result.length
+				} else {
+					let index = foundKeys[firstChar] - 1
+					result[index].values.push(user)
+				}
+			}
+
+			const sortedData = _.sortBy(result, 'key') || []
```
Comment (rakeshSgr) [inline src/services/mentees.js:915]: @adithyadinesh0412  add sort in query

**Pull request merged**